IPython/core/ultratb.py

```diff
@@ -907,7 +907,7 @@ def debugger(self, force: bool = False):
         fix that by hand after invoking the exception handler."""
 
         if force or self.call_pdb:
-            if self.pdb is None:
+            if self.debugger_cls:
                 self.pdb = self.debugger_cls()
```

Comment:
I don't think this is totally correct. pdb is supposed to be the instance, there is just another place that sets pdb without respecting `debugger_cls`.

Reply:
> there is just another place that sets pdb without respecting debugger_cls

which place exactly? ListTB?

Reply:
Sorry coming back to this.

Here in init:

https://github.com/ipython/ipython/blob/781e675a4bfafd50ebe7909d09bae290f92aad72/IPython/core/ultratb.py#L231-L234